ci: remove QT_QPA_PLATFORM=offscreen for non-Windows Qt tests

rokm opened the pull request:
Remove `QT_QPA_PLATFORM=offscreen` for non-Windows Qt tests.

Preliminary tests with PySide6 6.9.0 indicate that offscreen platform backend randomly breaks the QtWebEngine test on arm64 macOS (i.e., the `macos-14` runner); see [here](https://github.com/rokm/pyinstaller/actions/runs/14224706692) and [here](https://github.com/rokm/pyinstaller/actions/runs/14230408816).

And even on the remaining unaffected platform (linux), it likely makes more sense to test with the default platform rather than offscreen.

rokm committed:
ci: remove QT_QPA_PLATFORM=offscreen for non-Windows Qt tests

Remove `QT_QPA_PLATFORM=offscreen` for non-Windows Qt tests.

Preliminary tests with PySide6 6.9.0 indicate that offscreen
platform backend randomly breaks the QtWebEngine test on arm64
macOS (i.e., the `macos-14` runner).

And even on the remaining unaffected platform (linux), it likely
makes more sense to test with the default platform rather than
offscreen.

bwoodsend reviewed (approved):
Bit weird that we added that without removing `Xvfb`.

rokm commented:
> Bit weird that we added that without removing `Xvfb`.

We need Xvfb for other GUI frameworks, though, like tkinter and splash screen tests.

The fun part is that git blame states that I added this `QT_QPA_PLATFORM=offscreen`, and I completely forgot about that :) (and it's a bit ironic that it looks like it was partially done to mitigate issues with OpenGL on macOS runners at the time - https://github.com/pyinstaller/pyinstaller/commit/4b47eda847cb7a064f8c35a79b483bba3cf500ca / https://github.com/pyinstaller/pyinstaller/pull/6276).

rokm committed:
ci: ensure libxcb-cursor0 is installed on ubuntu runner

Needed by Qt6 when running with (default) `xcb` platform instead
of the `offscreen` one.
